.github/workflows/deploy.yml

```
@@ -1,9 +1,10 @@
 name: github pages
 
 on:
+  pull_request:
   push:
     branches:
-      - main  # Set a branch to deploy
+      - main
```

```suggestion
      - main
    workflow_dispatch:
```

This helps us run it manually with a button, very useful!